docs/docs/node-interface.md

```diff
@@ -11,11 +11,18 @@ The basic node data structure is as follows:
 id: String,
 children: Array[String],
 parent: String,
+// Reserved for plugins who wish to extend other nodes.
+pluginFields: Object,
 internal: {
   contentDigest: String,
   mediaType: String,
+  // A globally unique node type choosen by the plugin owner.
   type: String,
-  pluginName: String,
+  // The plugin which created this node.
+  pluginOwner: String,
+  // Stores which plugins created which fields.
+  fieldPluginOwners: Object,
```

Comment:
It's true that a name clash with `fields` is a little more likely to occur, but I would still choose a cleaner api over that.